nixos/modules/services/networking/nextdns.nix

```diff
@@ -0,0 +1,41 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nextdns;
+in {
+  options = {
+    services.nextdns = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether or not to enable the NextDNS DNS/53 to DoH Proxy service.";
+      };
+      arguments = mkOption {
+        type = types.separatedString " ";
+        example = "-config 10.0.3.0/24=abcdef";
+        default = "";
+        description = "Additional arguments to be passed to nextdns run.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.nextdns = {
+      description = "NextDNS DNS/53 to DoH Proxy";
+      serviceConfig = {
+        StartLimitInterval = 5;
+        StartLimitBurst = 10;
+        Environment = "SERVICE_RUN_MODE=1";
+        ExecStart = "${pkgs.nextdns}/bin/nextdns run ${config.services.nextdns.arguments}";
+        RestartSec = 120;
+        LimitMEMLOCK = "infinity";
```

Comment:
Why is this needed? A comment would be nice

Reply:
I have no idea. I copied the template from the application source code. I'll add a comment.

https://github.com/nextdns/nextdns/blob/628ea509eaaccd27adb66337db03e5b56f6f38a8/host/service/systemd/service.go